src/native/libs/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -139,6 +168,36 @@ int32_t CryptoNative_EvpDigestOneShot(const EVP_MD* type, const void* source, in
     return ret;
 }
 
+int32_t CryptoNative_EvpDigestXOFOneShot(const EVP_MD* type, const void* source, int32_t sourceSize, uint8_t* md, uint32_t len)
+{
+    ERR_clear_error();
+
+    if (type == NULL || sourceSize < 0 || md == NULL)
+    {
+        return 0;
+    }
+
+    EVP_MD_CTX* ctx = CryptoNative_EvpMdCtxCreate(type);
+
+    if (ctx == NULL)
+    {
+        return 0;
+    }
+
+    int32_t ret = EVP_DigestUpdate(ctx, source, (size_t)sourceSize);
+
+    if (ret != SUCCESS)
+    {
+        CryptoNative_EvpMdCtxDestroy(ctx);
+        return 0;
+    }
+
+    ret = CryptoNative_EvpDigestFinalXOF(ctx, md, len);
```

Comment:
Jeremy: Is it worth front-loading a check for EVP_DigestFinalXOF existing so that we don't do a bunch of work before realizing we can't finish?
Kevin: Well, we'll only call this function when it's SHAKE128/256, and in that case we either already threw that it wasn't working, or we messed up and we're going to pass an invalid type so Create would have failed, or everything's going to work.
Jeremy: Right. Nevermind.